### DISCUSSION TITLE
Multiple jdtls lsp client attached for each maven sub-project module

### DISCUSSION THREAD

**Discussion by jeusdi** [mfussenegger/nvim-jdtls#635] [Q&A]
I'm using lazyvim.

I'm working into a maven multi-module java project:

```
.
├── pom.xml
├── application
│   ├── pom.xml
│   └── src
├── domain
│   ├── pom.xml
│   └── src
└── infrastructure
    ├── pom.xml
    └── src
```

However, every time I open a file of any concrete project, a new `jdtls` lsp is laucnhed.

I've took a look to [lazyvim java lang extra documentation](https://www.lazyvim.org/extras/lang/java), but I don't quite figure out how to deal with it.

I'd like that when I open into any class of my multimodule maven project, only one lst client is attached to a single jdtls lsp.

Any ideas?

**Comment by azthec** [comment]
You seem to have the same issue I was having, an easy way to tell is if you can't access say your domain imports from your application module, or you can't find references across modules.

If so what is happening is that `nvim-jdtls` `root directory` is not being set properly, you likely have this happening automatically via `root_dir = jdtls.setup.find_root(root_files)` or `root_dir = vim.fs.dirname(vim.fs.find({'gradlew', '.git', 'mvnw'}, { upward = true })[1])`.

It seems that by default this matches on one file found, and then recursively goes up, so it is treating each of your modules as completely separate projects.
